src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -62,7 +62,10 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(UseCurrentRuntimeIdentifier)' == 'true'">
-    <RuntimeIdentifier>$(NETCoreSdkRuntimeIdentifier)</RuntimeIdentifier>
+    <RuntimeIdentifier Condition="$(NETCoreSdkRuntimeIdentifier.EndsWith('arm64')) and
+                                  $(NETCoreSdkRuntimeIdentifier.StartsWith('osx')) and
+                                  '$(_TargetFrameworkVersionWithoutV)' &lt; '6.0'">$([System.String]::Copy('$(NETCoreSdkRuntimeIdentifier)').Replace("arm64", "x64"))</RuntimeIdentifier>
+    <RuntimeIdentifier Condition="'$(RuntimeIdentifier)' == ''">$(NETCoreSdkRuntimeIdentifier)</RuntimeIdentifier>
```

Comment:
@wli3 or @dsplaisted could you please review this piece? The goal is when running on osx-arm64 and working with downlevel app (TFM < 6.0) to switch to osx-x64 since the first version which supports osx-arm64 is .NET 6 (downlevel versions don't have support and only run on x64, also macOS arm64 can emulate x64, so it's the right thing to do to run the app as x64).

The main concern we had was how to construct the new RID - specifically if it should maintain the potential OS version and such in the RID or not.

Reply:
This will only apply for projects that opt in to `UseCurrentRuntimeIdentifier`, or when `--use-current-runtime` is passed on the command line.  Is that the intention?

I thought that we were going to use this type of logic for the the `DefaultAppHostRuntimeIdentifier`.

It's not great that we have to do string manipulation to check the RuntimeIdentifier, but I don't think there's a better option.

I do think when setting the value you don't need to use `String.Copy`.  You can just call replace directly, String.Copy is a workaround to call replace on item metadata, and shouldn't be needed for a property value.

Reply:
I don't know what's the difference between `UseCurrentRuntimeIdentifier` and `DefaultAppHostRuntimeIdentifier`. The intent of this change is that typing `dotnet run` on a 3.1 project when running on macOS arm64 SDK will produce macOS x64 app output (apphost) and thus will be able to execute the app. This then also means that `dotnet build` should behave that way.

In short - SDK should not default to the RID of the SDK when targeting downlevel TFMs and such TFMs are not supported on the RID. It should default to whatever RID is actually supported for the TFM on the current machine.

Reply:
`DefaultAppHostRuntimeIdentifier` is the primary property you should be concerned with.  That's what determines which apphost gets used if a `RuntimeIdentifier` (or `AppHostRuntimeIdentifier`) isn't otherwise specified.

`UseCurrentRuntimeIdentifier` is a new-ish option to allow building a self-contained app without explicitly specifying the RuntimeIdentifier.  See #14093.  It is probably a good idea for `UseCurrentRuntimeIdentifier` to switch from arm64 to x64 when targeting .NET 5.0 or earlier, but that's not the mainline scenario you're trying to address.

Reply:
That makes sense. I've changed things so we modify the `DefaultAppHostRuntimeIdentifier`.

Reply:
If `UseCurrentRuntimeIdentifier` isn't set to `DefaultAppHostRuntimeIdentifier` I think we should set that too. Otherwise I think that scenario will be broken for Mac.